src/lua/utils.c

```diff
@@ -696,7 +696,7 @@ void cord_on_yield(void)
 		 * world and one can obtain the corresponding Lua
 		 * coroutine from the fiber storage.
 		 */
-		struct lua_State *L = fiber()->storage.lua.stack;
+		struct lua_State *L = gco2th(gcref(g->cur_L));
```

Comment:
This part is in review in PR #8048 (against `master`). I left this comment as reminder to wait for it and to add 'cherry-picked from <...>' later into the commit message.